test/statetest/statetest_loader.cpp

```diff
@@ -0,0 +1,199 @@
+        return EVMC_PARIS;
+    throw std::invalid_argument{"unknown revision: " + std::string{s}};
+}
+}  // namespace
+
+static void from_json(const json::json& j, TestMultiTransaction& o)
+{
+    if (j.contains("gasPrice"))
+    {
+        o.kind = state::Transaction::Kind::legacy;
+        o.max_gas_price = from_json<intx::uint256>(j.at("gasPrice"));
+        o.max_priority_gas_price = o.max_gas_price;
+    }
+    else
+    {
+        o.kind = state::Transaction::Kind::eip1559;
+        o.max_gas_price = from_json<intx::uint256>(j.at("maxFeePerGas"));
+        o.max_priority_gas_price = from_json<intx::uint256>(j.at("maxPriorityFeePerGas"));
+    }
+    o.sender = from_json<evmc::address>(j.at("sender"));
+    if (!j.at("to").get<std::string>().empty())
+        o.to = from_json<evmc::address>(j["to"]);
+
+    for (const auto& j_data : j.at("data"))
+        o.inputs.emplace_back(from_json<bytes>(j_data));
+
+    if (j.contains("accessLists"))
+    {
+        for (const auto& j_access_list : j["accessLists"])
+            o.access_lists.emplace_back(from_json<state::AccessList>(j_access_list));
+    }
+
+    for (const auto& j_gas_limit : j.at("gasLimit"))
+        o.gas_limits.emplace_back(from_json<int64_t>(j_gas_limit));
+
+    for (const auto& j_value : j.at("value"))
+        o.values.emplace_back(from_json<intx::uint256>(j_value));
+}
+
+static void from_json(const json::json& j, TestMultiTransaction::Indexes& o)
+{
+    o.input = j.at("data").get<size_t>();
+    o.gas_limit = j.at("gas").get<size_t>();
+    o.value = j.at("value").get<size_t>();
+}
+
+static void from_json(const json::json& j, StateTransitionTest::Case::Expectation& o)
+{
+    o.indexes = j.at("indexes").get<TestMultiTransaction::Indexes>();
+    o.state_hash = from_json<hash256>(j.at("hash"));
+    o.logs_hash = from_json<hash256>(j.at("logs"));
+    o.exception = j.contains("expectException");
+}
+
+static void from_json(const json::json& j, StateTransitionTest& o)
+{
+    const auto& j_t = j.begin().value();  // Content is in a dict with the test name.
+
+    for (const auto& [j_addr, j_acc] : j_t.at("pre").items())
+    {
+        const auto addr = from_json<address>(j_addr);
+        auto& acc = o.pre_state.create(addr);
+        acc.balance = from_json<intx::uint256>(j_acc.at("balance"));
+        acc.nonce = from_json<uint64_t>(j_acc.at("nonce"));
+        acc.code = from_json<bytes>(j_acc.at("code"));
+
+        for (const auto& [j_key, j_value] : j_acc.at("storage").items())
+        {
+            auto& slot = acc.storage[from_json<bytes32>(j_key)];
+            const auto value = from_json<bytes32>(j_value);
+            slot.original = value;
+            slot.current = value;
+        }
+    }
+
+    o.multi_tx = j_t.at("transaction").get<TestMultiTransaction>();
+
+    o.block = from_json<state::BlockInfo>(j_t.at("env"));
+
+    for (const auto& [rev_name, expectations] : j_t.at("post").items())
+        o.cases.push_back({to_rev(rev_name),
```

Comment:
I will try, but this may not work because this class does not have explicit constructor. I believe this is fixed in C++20 or C++23 where `S(a, b)` is valid initialization for an aggregate `S`.